### PR TITLE
Single well state3

### PR DESCRIPTION
**Pull request by joakim-hove**
Part of #3480

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

File: opm/simulators/wells/SingleWellState.hpp [hunk 1]
```diff
@@ -20,20 +20,28 @@
...
 public:
-    SingleWellState(bool is_producer, double temp);
+    SingleWellState(bool is_producer, std::size_t num_phases, double temp);
 
     Well::Status status{Well::Status::OPEN};
     bool producer;
     double bhp{0};
     double thp{0};
-    double temperature{};
+    double temperature;
```
Comment (bska) [inline opm/simulators/wells/SingleWellState.hpp:39]: This makes `temperature` undefined in the `SingleWellState` unless a call site explicitly assigns a value.  Please revert.

**Review by bska — CHANGES_REQUESTED** [review]
Most of this looks to be a mechanical API change, but there's one piece of the PR that causes undefined behaviour.  Please fix.

File: opm/simulators/wells/SingleWellState.hpp (hunk 1 above)
Comment (joakim-hove) [reply opm/simulators/wells/SingleWellState.hpp:39]: I'll revert sure enough, not intentional in any way. That said I do think the fringe feature temperature gets way to much attention/focus in the well code.

Comment (joakim-hove) [reply opm/simulators/wells/SingleWellState.hpp:39]: Ehhh - second thought; the change to uninitialized was actually semi-intentional - to highlight the fact that the temperature value is just garbage. With `double temperature{}` it is well defined from a C++ standpoint, but from a physics/reservoir it still just garbage - although garbage.

Anyway - I'll put that :horse: back in the stable and revert!

**Comment by joakim-hove** [comment]
jenkins build this please

File: opm/simulators/wells/SingleWellState.hpp (hunk 1 above)
Comment (bska) [reply opm/simulators/wells/SingleWellState.hpp]: > the change to uninitialized was actually semi-intentional - to highlight the fact that the temperature value is just garbage.

I appreciate that, but by making the data member uninitialised you also create undefined behaviour in any operation that works on the structure as a whole.

> With `double temperature{}` it is well defined from a C++ standpoint

And that's needed.

**Review by bska — APPROVED** [review]
Thanks for the updates.  I'll merge into master.

**Pull request merged**